changelogs/fragments/8906.yml

```diff
@@ -0,0 +1,2 @@
+fix:
+- Index pattern fetch error in discover dataset modal ([#8906](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8906))
```

Comment:
Although this PR could fix https://github.com/opensearch-project/OpenSearch-Dashboards/issues/8891, but this is not the direct root cause, so I would suggest to change to `refactor: refactor the bulk_get handler in permission wrapper when item has permission error`